mne/io/meas_info.py

```diff
@@ -206,6 +206,19 @@ def _anonymize(self):
         if self.get('subject_info') is not None:
             del self['subject_info']
 
+    def _check_consistency(self):
+        """Run some self-consistency checks"""
+        missing = [bad for bad in self['bads'] if bad not in self['ch_names']]
+        if len(missing) > 0:
+            raise RuntimeError('bad channel(s) %s marked do not exist in info'
+                               % (missing,))
+        chs = [ch['ch_name'] for ch in self['chs']]
+        if len(self['ch_names']) != len(chs) or any(
+                ch_1 != ch_2 for ch_1, ch_2 in zip(self['ch_names'], chs)) or \
+                self['nchan'] != len(chs):
+            raise RuntimeError('info channel name inconsistency detected, '
+                               'please notify mne-python developers')
+
```

Comment:
add a check that a sfreq is a float as well as low_pass and high_pass if not None